scripts/sbatch.py

```diff
@@ -0,0 +1,476 @@
+MAUI_SLUM_CONF = "/scale_wlg_persistent/filesets/home/slurm/maui/etc/opt/slurm/slurm.conf"
+MAUI_ANCIL_SLUM_CONF = "/scale_wlg_persistent/filesets/home/slurm/maui_ancil/etc/opt/slurm/slurm.conf"
+
+SBATCH_BIN = "/opt/slurm/18.08.7/bin/sbatch"
+
+
+#TODO: classes can be included into qcore libs ( if qcore become a required dependency)
+class HPC(Enum):
+    #since default value for multithread is True, the numbers are logical cores
+    maui = (0,"maui")
+    maui_ancil = (1,"maui_ancil")
+    mahuika = (2,"mahuika",)
+
+    def __new__(cls, value, id):
+        obj = object.__new__(cls)
+        obj._value_ = value
+        obj.id = id
+        return obj
+
+
+#Trace-able resources and the default weight value for Billing
+class TRESWeight(Enum):
+    cpu = (0,"CPU", 1.0)
+    mem = (1,"Mem", 0.0)
+    gpu = (2,"gpu", 0.0)
+
+    def __new__(cls, value, id, weight):
+        obj = object.__new__(cls)
+        obj._value_ = value
+        obj.id = id
+        obj.default_weight = weight
+        return obj
+
+
+# levels of parameters in config
+class varLevel(Enum):
+    specific = 0,'specific'
+    default = 1,'DEFAULT'
+    root = 2,'root'
+    def __new__(cls, value, id):
+        obj = object.__new__(cls)
+        obj._value_ = value
+        obj.id = id
+        return obj 
+
+
+# can be moved to qcore const
+class SlurmHeader(Enum):
+    account = (0,"account", str, 'nesi00213')
+    partition = (1,"partition", str, None)
+    ntasks = (2,"ntasks", int, 1)
+    ntasks_per_core = (3,"ntasks-per-core", int, 1)
+    cpus_per_task = (4,"cpus-per-task", int, 1)
+    mem_per_cpu = (5,"mem-per-cpu", str, None)
+    ntasks_per_node = (6, "ntasks-per-node", int, 0)
+    time = (7, "time", int, 0) #in seconds
+    #flags
+    nomultithread = (8,"hint=nomultithread", bool, False)
+    exclusive = (9,"exclusive", bool, False)
+
+    def __new__(cls, value, id,type, default):
+        obj = object.__new__(cls)
+        obj._value_ = value
+        obj.id = id
+        obj.type = type
+        obj.default = default
+        return obj
+
+
+#TODO:nest partition classes with HPC classes so no double define
+class MahuikaPartition(Enum):
+    large = (0, "large", 72, 108*MB_PER_GB)
+    long = (1, "long", 72, 108*MB_PER_GB)
+    perpost = (2, "prepost", 72, 480*MB_PER_GB)
+    bigmem = (3, "bigmem", 72, 480*MB_PER_GB)
+    hugmem = (4, "hugemem", 128, 4000*MB_PER_GB)
+    ga_bigmem = (5, "ga_bigmem", 72, 480*MB_PER_GB)
+    ga_hugemem = (6, "ga_hugemem", 128, 4000*MB_PER_GB)
+    gpu = (7, "gpu", 8, 108*MB_PER_GB)
+    igpu = (8, "igpu", 8, 108*MB_PER_GB)
+
+    def __new__(cls, value, id, cpu_per_node, mem_per_node):
+        obj = object.__new__(cls)
+        obj._value_ = value
+        obj.id = id
+        obj.cpus = cpu_per_node
+        obj.mem = mem_per_node
+        obj.config_path = MAHUIKA_SLURM_CONF
+        obj.hpc = HPC.mahuika.value
+        return obj
+
+
+class MauiPartition(Enum):
+    nesi_research = (0, "nesi_research", 80, 160*MB_PER_GB)
```

Comment:
Use a named tuple or class, or split into attributes.

Reply:
there is some automated value assign in the __new__, so if its changed to named tuple than there will be a lot of repeittive values. (i.e. hpc num and conf for each partition)

Reply:
A struct/class is probably better then. (just to give the parameters some naming rather than 1, 2, 3)

Reply:
but then iterations will need to use dictionaries instead of just a list/array?